scripts/extensions/sams/src/utils/api.ts

```diff
@@ -0,0 +1,32 @@
+// External Modules
+import {generate} from 'json-merge-patch';
+
+// Types
+import {IBaseRestApiResponse} from 'superdesk-api';
+
+export function generatePatch<T extends IBaseRestApiResponse>(original: T, updates: Partial<T>): Partial<T> {
```

Comment:
I added this function because the code in the core was messy and not fixable in a reasonable time. I don't see a reason to carry this over to the new code. It should be fine for you to use `generate` directly. The only issue is that the library has poor types so maybe it might be worth creating a tiny wrapper just for types
```
export function generatePatch<T>(obj1: Partial<T>, obj2: Partial<T>): Partial<T> {
    return (generate(obj1, obj2) ?? {}) as Partial<T>
}
```